pkg/engine/task/task_pipe.go

```diff
@@ -0,0 +1,452 @@
+// as ConfigMap data key. ConfigMap name will be of the form <instance>.<plan>.<phase>.<step>.<task>-<PipeFile.Key>
+func pipeConfigMap(pf PipeFile, data []byte, meta renderer.Metadata) (string, error) {
+	name := PipeArtifactName(meta, pf.Key)
+	key := path.Base(pf.File)
+
+	configMap := corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		BinaryData: map[string][]byte{key: data},
+	}
+
+	b, err := yaml.Marshal(configMap)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal pipe configMap for pipe file %s: %v", pf.File, err)
+	}
+
+	return string(b), nil
+}
+
+// untarFile extracts a tar file from the passed reader using passed file name.
+func untarFile(fs afero.Fs, r io.Reader, fileName string) (err error) {
+	tr := tar.NewReader(r)
+
+	for {
+		header, err := tr.Next()
+		if err != nil {
+			if err != io.EOF {
+				return err
+			}
+			break
+		}
+
+		// the target location of the file. tar strips the leading "/" however, we treat the pipe file path
+		// as a key to the underlying data (otherwise we'll have to start splitting paths). To avoid all
+		// the complexity and because we only extract one file here, the path is taken from the PipeFile configuration
+		target := fileName
+
+		// check the file type
+		switch header.Typeflag {
+		// if it's a file create it
+		case tar.TypeReg:
+			f, err := fs.OpenFile(target, os.O_CREATE|os.O_RDWR, os.FileMode(header.Mode))
+			if err != nil {
+				return err
+			}
+
+			// copy over contents
+			if _, err := io.Copy(f, tr); err != nil {
+				return err
+			}
+
+			// manually close here after each file operation; deferring would cause each file close
+			// to wait until all operations have completed.
+			f.Close() // nolint
+
+		default:
+			fmt.Printf("skipping %s because it is not a regular file or a directory", header.Name)
+		}
+	}
+
+	return nil
+}
+
+// PipePodName returns a deterministic name for a pipe pod
+func PipePodName(meta renderer.Metadata) string { return name(meta, "pipepod") }
+
+// PipeArtifactName returns a deterministic name for pipe artifact (ConfigMap, Secret)
+func PipeArtifactName(meta renderer.Metadata, key string) string { return name(meta, key) }
+
+var (
+	alnum = regexp.MustCompile(`[^a-z0-9]+`)
+)
+
+// name returns a deterministic names for pipe artifacts (Pod, Secret, ConfigMap) in the form:
+// <instance>.<plan>.<phase>.<step>.<task>.<suffix> All non alphanumeric characters are removed.
+// A name for e.g a ConfigMap has to match a DNS-1123 subdomain, must consist of lower case alphanumeric
+// characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com',
+// regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
+func name(meta renderer.Metadata, suffix string) string {
```

Comment:
Nit: Use a more descriptive name for this function, how about `pipeName`?

Reply:
`pipeName` sound like a name for the `pipe`.  seems like an overkill for a private method which is called from `PipePodName` and `PipeArtifactName`. but if you have a short and concise name - I'm open ;)